src/tile.cpp

```diff
@@ -742,27 +744,28 @@ ReturnValue Tile::queryRemove(const Thing& thing, uint32_t count, uint32_t flags
 	return RETURNVALUE_NOERROR;
 }
 
-Tile* Tile::queryDestination(int32_t&, const Thing&, Item** destItem, uint32_t& flags)
+std::shared_ptr<Thing> Tile::queryDestination(int32_t&, std::shared_ptr<const Thing>, std::shared_ptr<Item>& destItem,
+                                              uint32_t& flags)
 {
-	Tile* destTile = nullptr;
-	*destItem = nullptr;
+	std::shared_ptr<Tile> destTile = nullptr;
+	destItem = nullptr;
 
 	if (hasFlag(TILESTATE_FLOORCHANGE_DOWN)) {
 		uint16_t dx = tilePos.x;
 		uint16_t dy = tilePos.y;
 		uint8_t dz = tilePos.z + 1;
 
-		Tile* southDownTile = g_game.map.getTile(dx, dy - 1, dz);
+		auto southDownTile = g_game.map.getTile(dx, dy - 1, dz);
 		if (southDownTile && southDownTile->hasFlag(TILESTATE_FLOORCHANGE_SOUTH_ALT)) {
 			dy -= 2;
 			destTile = g_game.map.getTile(dx, dy, dz);
 		} else {
-			Tile* eastDownTile = g_game.map.getTile(dx - 1, dy, dz);
+			auto eastDownTile = g_game.map.getTile(dx - 1, dy, dz);
 			if (eastDownTile && eastDownTile->hasFlag(TILESTATE_FLOORCHANGE_EAST_ALT)) {
 				dx -= 2;
 				destTile = g_game.map.getTile(dx, dy, dz);
 			} else {
-				Tile* downTile = g_game.map.getTile(dx, dy, dz);
+				auto downTile = g_game.map.getTile(dx, dy, dz);
```

Comment:
Could this be `const auto`
```suggestion
				const auto downTile = g_game.map.getTile(dx, dy, dz);
```